modules/ROOT/assets/attachments/sg.yaml

```diff
@@ -696,6 +706,400 @@ properties:
+                    type: string
+                    default: None - this is mandatory
+                    description: |2
+                      **About**
+
+                      The **`remote`** property represents a database endpoint for the remote Sync Gateway cluster.
+                      That is, it identifies the remote cluster that is the subject of this replication's push, pull or push_and_pull action.
+
+                      **Options**
+
+                      The format can be one of --
+                      - a string containing a valid URL for a (remote) Sync Gateway database.
+                      - an object whose url property contains the Sync Gateway database URL.
+
+                      **Behavior**
+
+                      Dependent upon the setting of the [direction](./../refer/config-properties.html#databases-this_db-replications-direction) configuration property.
+
+                      If direction is :
+                      - `direction=pull`, then `remote` defines the remote cluster *from* which data is pulled
+                      - `direction=push`, then `remote` defines the remote cluster *to* which data is pushed
+                      - `direction=push_and_pull`, then `remote` defines the remote cluster *to* which data is pushed.
+
+                      **Constraints**
+
+                      - You must specify the 'remote' database's url even if it is located on the same cluster as the replication's database.
+                      - OPTIONAL for stops and removes
+
+
+                  replication_id:
+                    type: string
+                    description: |2
+                      **About**
+
+                      The *replication_id* property specifies either:
+                      - For NEW replications, the ID to be assigned to the the replication. If no *replication_id* is specified, Sync Gateway will assign a random UUID to new replications.
+                      - For existing replications, this is the ID of the required replication.
+                      - If **cancel=true**, this is the id of the active replication task to be cancelled.
+
+                      **Constraints**
+
+                      If this is specified in the body of a POST or PUT request then it must be the same value as specified in the request URL.
+
+                  state:
+                    type: string
+                    default: Running
+                    description: |2
+                      **About**
+
+                      The optional `state` configuration property is used to specify that the replication must be launched in 'Stopped' mode
+
+                      **Behavior**
+
+                      All replications are configured to start on Sync Gateway launch. So, if omitted, the state defaults to 'Running'.
+
+                      **Constraints*
+
+                      Replications created prior to version 2.8 will all default to a state of 'Running'.
+
```

Comment:
'username' and 'password' properties have been added to the replication config.  They can be used to specify the remote user, as an alternative to specifying them on the query string in the `remote` definition.